garden-backend-service/src/models/mdf_dataset.py

```diff
@@ -0,0 +1,20 @@
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Mapped, mapped_column
+from src.models.base import Base
+
+
+class Dataset(Base):
+    __tablename__ = "mdf-datasets"
+    id: Mapped[str] = mapped_column(primary_key=True)  # mdf SI subject id
+    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
```

Comment:
this is for ownership, right? take a look at the extra ORM stuff we do for an `owner` field on gardens and entrypoints, should make the actual code for the queries a little easier to write